src/task_monitor/tasks/process_identities.rs

```diff
@@ -13,17 +15,20 @@ use tracing::{debug, error, info, instrument, warn};
 use crate::contracts::{IdentityManager, SharedIdentityManager};
 use crate::database::Database;
 use crate::identity_tree::{
-    AppliedTreeUpdate, Intermediate, TreeVersion, TreeVersionReadOps, TreeWithNextVersion,
+    AppliedTreeUpdate, Hash, Intermediate, TreeVersion, TreeVersionReadOps, TreeWithNextVersion,
 };
 use crate::prover::identity::Identity;
 use crate::prover::map::ReadOnlyInsertionProver;
+use crate::prover::{Proof, Prover, ReadOnlyProver};
 use crate::task_monitor::{PendingBatchSubmission, TaskMonitor};
 use crate::utils::async_queue::AsyncQueue;
 
 /// The number of seconds either side of the timer tick to treat as enough to
 /// trigger a forced batch insertion.
 const DEBOUNCE_THRESHOLD_SECS: u64 = 1;
 
+const RECOVERY_WAIT_TIME: Days = Days::new(7);
```

Comment:
I think ideally we should read this from chain. We could even read it on every batch I think - we're producing them pretty infrequently.